### PR TITLE
Moved sync GitHub account control to frontend

### PR DESCRIPTION
**Pull request by roll**
- fixes [partially] #137 
- simplified vue component tag names

---

It covers github sync account without activate/deactivate logic. Also `GitHubSettings` component's tests could be used as role model for testing interactive vue components (including network dependent).

To finish #137 there is a relatively small step of moving activate/deactivate clicks to frontend. It will be addressed in separate PR.

### CODE REVIEW

**Comment by roll** [comment]
@amercader 
please take a look

File: goodtablesio/integrations/github/models/repo.py [hunk 1]
```diff
@@ -7,6 +7,13 @@ class GithubRepo(Source):
         'polymorphic_identity': 'github'
     }
 
+    def to_api(self):
```
Comment (amercader) [inline goodtablesio/integrations/github/models/repo.py]: I guess the point of this and not use `to_json()` is to control exactly what gets sent, +1

File: goodtablesio/integrations/github/tasks/repos.py [hunk 2]
```diff
@@ -12,6 +12,7 @@ def sync_user_repos(user_id):
     """
     user = database['session'].query(User).get(user_id)
 
+    # TODO: handle no token situation
```
Comment (amercader) [inline goodtablesio/integrations/github/tasks/repos.py]: Abort with an exception :)

File: goodtablesio/integrations/github/models/repo.py (hunk 1 above)
Comment (roll) [inline goodtablesio/integrations/github/models/repo.py]: Yes some sensitive data added later could be leaked otherwise

**Comment by roll** [comment]
- [x] Can we disable the "Sync Account" button while syncing is in progress?
- [x] I think we can slow down the ajax calls to one every 3-5 seconds

**Pull request merged**